### ISSUE TITLE
InvalidStaticInvocation false positive: static call to inaccessible non-static method routes to __callStatic at runtime

### ISSUE THREAD

**Issue by alies-dev** [vimeo/psalm#11876]
https://psalm.dev/r/7589c15285

### Summary

`InvalidStaticInvocation` is emitted for a static call to a non-static method **even when that method is inaccessible from the calling scope and the class declares `__callStatic`**. In that situation PHP does not call the method directly — it routes the call to `__callStatic()` — so the code is runtime-valid and the error is a false positive.

```php
<?php

class Repository
{
    public static function __callStatic(string $name, array $args): mixed
    {
        return (new static())->$name(...$args);
    }

    protected function find(int $id): string
    {
        return "found $id";
    }
}

// Runtime-valid: find() is protected, so PHP routes this to
// __callStatic() -> (new static())->find(1). Returns "found 1".
Repository::find(1);
```

### Expected

No issue on `Repository::find(1)`: the call is dispatched through `__callStatic` at runtime, exactly as it would be for a method name that does not exist at all (Psalm already routes those to the magic-method path).

### Actual

Two errors on that line:

- `InvalidStaticInvocation` — `Method Repository::find is not static, but is called statically`
- `InaccessibleMethod` — `Cannot access protected method Repository::find`

Note the second error: Psalm already determines the method is **inaccessible**, yet still treats the call as a direct invocation of that method. The two diagnostics contradict each other — if the method cannot be accessed, the static call does not invoke it; PHP forwards to `__callStatic`.

### The PHP dispatch rule (accessibility-gated)

A static call to a non-static method raises `Error` **only when the method is accessible** from the calling scope. When it is inaccessible, PHP routes to `__callStatic`:

```php
class C {
    public static function __callStatic($n, $a) { return "__callStatic:$n"; }
    public    function pub()  {}
    protected function prot() {}
    private   function priv() {}
}

C::pub();   // Error: Non-static method C::pub() cannot be called statically
C::prot();  // "__callStatic:prot"  (inaccessible from outside -> __callStatic)
C::priv();  // "__callStatic:priv"  (inaccessible from outside -> __callStatic)
```

From inside the class / a subclass the protected/private method *is* accessible, so PHP raises the same `Error` there — confirming the rule is purely accessibility-based (verified on PHP 8.4.22).

This means the fix is **accessibility-gated, not a blanket suppression**: the `public` case (`C::pub()` above) must keep emitting `InvalidStaticInvocation` — that one is a true positive, because PHP really does refuse it.

### Source

`Psalm\Internal\Analyzer\MethodAnalyzer::checkStatic` decides solely on `is_static`, with no accessibility or `__callStatic` consideration:

```php
$storage = $codebase_methods->getStorage($method_id, $with_pseudo);

if (!$storage->is_static) {
    // ...
    } else {
        if (IssueBuffer::accepts(
            new InvalidStaticInvocation(
                'Method ' . $codebase_methods->getCasedMethodId($method_id) .
                    ' is not static, but is called statically',
                $code_location,
            ),
```

Suggested direction: when the resolved method exists but is **inaccessible from the calling context** and the class declares `__callStatic`, defer to the magic-static-method path (as already happens for non-existent methods) instead of emitting `InvalidStaticInvocation` / `InaccessibleMethod`.

### Workaround status

Adding `@method static string find(int $id)` to the class supplies the return type, but **does not** silence `InvalidStaticInvocation`: the real (inaccessible) method shadows the pseudo-method, so the false positive remains.

### PHPStan, for reference

PHPStan (level max) reports the same call as `Static call to instance method Repository::find()` (`method.staticCall`); it does not special-case the `__callStatic` fallback in core either. This is shared behaviour, not a Psalm-specific regression — included only for completeness. The PHP runtime semantics above are the authority for whether the call is valid.

### Real-world motivation

Laravel 12's documented query-scope syntax uses the `#[Scope]` attribute on a **protected** model method, e.g. `protected function popular(Builder $q)`, invoked as `Order::popular()` (https://laravel.com/docs/12.x/eloquent#local-scopes). At runtime that routes through `Model::__callStatic` → `static::query()->popular()`. Every such call currently trips `InvalidStaticInvocation`, and it cannot be fixed cleanly downstream: `pseudo_static_methods` cannot preempt the real method's storage, and flipping `is_static` on the method breaks `$this` inside its body. (Context: https://github.com/psalm/psalm-plugin-laravel)

### Version

Psalm `7.0.0-beta19@7e751c06`. Also reproduces on the latest playground build (snippet link above).
